Kernel/FileSystem/Ext2FS/FileSystem.cpp

```diff
@@ -664,7 +663,13 @@ void Ext2FS::flush_writes()
         });
     }
 
-    BlockBasedFileSystem::flush_writes();
+    auto result = BlockBasedFileSystem::flush_writes();
+    if (result.is_error()) {
+        dbgln("Ext2FS[{}]::flush_writes(): Failed to flush writes: {}", BlockBasedFileSystem::fsid(), result.error());
+        return EIO;
```

Comment:
Should we return the same error we got from the base class? i.e. `return result.error();`